src/main/java/org/hawkular/client/android/backend/BackendClient.java

```diff
@@ -290,6 +287,11 @@ public void getTriggers(@NonNull Callback<List<Trigger>> callback) {
         getPipe(pipeName).read(getFilter(uri), callback);
     }
 
+    @SuppressWarnings("unchecked")
+    private <T> void savePipe(String pipeName, Object object, Callback<List<T>> callback, Class pipeClass) {
+        getPipe(pipeName).save(pipeClass.cast(object), callback);
```

Comment:
I think we don't need `pipeClass` as a method parameter and also `pipeClass.cast(object)`

Any special reason to use that? You had some problem without it?